src/Features/CSharp/Portable/ExtractMethod/CSharpMethodExtractor.CSharpCodeGenerator.cs

```diff
@@ -716,6 +716,70 @@ static bool ReturnOperationBelongsToMethod(SyntaxNode returnOperationSyntax, Syn
 
                     return enclosingMethod == methodSyntax;
                 }
+
+                async Task<SemanticDocument> CheckReturnOperations<TDeclarationNode>(
+                    TDeclarationNode node,
+                    OperationStatus<IMethodSymbol> methodSymbolResult,
+                    SemanticDocument originalDocument,
+                    CancellationToken cancellationToken)
+                    where TDeclarationNode : SyntaxNode
+                {
+                    var semanticModel = originalDocument.SemanticModel;
+
+                    var methodOperation = semanticModel.GetOperation(node, cancellationToken);
+                    var returnOperations = methodOperation.DescendantsAndSelf().OfType<IReturnOperation>();
+
+                    foreach (var returnOperation in returnOperations)
+                    {
+                        // If the return statement is located in a nested local function or lambda it
+                        // shouldn't contribute to the nullability of the extracted method's return type
+                        if (!ReturnOperationBelongsToMethod(returnOperation.Syntax, methodOperation.Syntax))
+                        {
+                            continue;
+                        }
+
+                        var syntax = returnOperation.ReturnedValue?.Syntax ?? returnOperation.Syntax;
+                        var returnTypeInfo = semanticModel.GetTypeInfo(syntax, cancellationToken);
+                        if (returnTypeInfo.Nullability.FlowState == NullableFlowState.MaybeNull)
+                        {
+                            // Flow state shows that return is correctly nullable
+                            return await base.UpdateMethodAfterGenerationAsync(originalDocument, methodSymbolResult, cancellationToken).ConfigureAwait(false);
+                        }
+                    }
+
+                    return null;
```

Comment:
did this loggic basically stay the same? (just extracted into a local function?).

Reply:
Looks the same to me, just now doesn't require a specific SyntaxNode type